decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -47,133 +84,207 @@ def __init__(self, lifter):
             FloatType: self._lift_basic_type,
             BoolType: self._lift_basic_type,
             VoidType: self._lift_void_type,
-            ArrayType: self._lift_constant_type,
+            ArrayType: self._lift_array_type,
             PointerType: self._lift_pointer_type,
-            NamedTypeReferenceType: self._lift_named_type_ref,  # Lift DataVariable with type NamedTypeRef
+            NamedTypeReferenceType: self._lift_named_type_ref,
         }
+        self._lifted_globals: dict[int, GlobalVariable] = {}
+        self._view: Optional[BinaryView] = None
+        self._callers: list[int] = []
```

Comment:
It could be done, but then the list must be part of the function parameters of each function.

I think it should work when i change the all signatures to use `**kwargs` and pass it as a named argument.
(Will still blow up the signature a bit, but at least the handlers which don't need the `._callers` don't have it in their signature.

Would also fix the problem with the public function which uses the private `._callers`

Reply:
Ok, it's good to know that callers is supposed to work like a parameter. If we decided that `_callers` stays a field, this should definitely be documented.

However, I still believe callers should be a real parameter. And if it becomes a parameter, we should strongly consider to either use tuples or to copy the list before changing it. Because otherwise we would wrongly detect cycles in trees of Global Variables with duplicate nodes. (Because the appended entries would propagate back).

Reply:
Callers is now a real parameter (see above).